[#7] Adaptive Styling

ChristofLee opened the pull request:
**THIS IS A WORK IN PROGRESS**
I require some help before this is finished.
- How many icons will be displayed?
- Is there a set number of icons displayed (e.g. 4) or will users be able to add more icons?
- if the above is true, will there be a button to add more icons?
- Am I permitted to edit the classes and markup here? e.g. add <a> or <button> tags, use BEM classes
- Is this intended to be styled _easily_ when added into a site? or should our styles take priority? e.g. color, font

## Summary
Some very basic styles for the emojion bar including animated hover effects.

- Responsive styles for the example.html doc
- Add style function to the pipe
- Load in some styles for the emojion bar
- BUG: Fix container being added to <style> tag

## Screenshots
Example of hover styles:
<img width="338" alt="hover-styles" src="https://cloud.githubusercontent.com/assets/4659534/25764776/68bea758-31e1-11e7-832b-1e4c4b09d269.png">


## Checklist
- [ ] I have updated the documentation to reflect my changes
- [ ] I have added tests to cover my changes (if relevant)
- [ ] All new and existing tests passed

Commit:
Fix issue where container was added to <style>

- Containers added to all id’s
- Formulate new id for <style> tag not using EMOJION_ID
- Create EMOJION_NAMESPACE, purposely different to EMOJION_ID

Commit:
Add function to style emojion bars

- Use style id const
- Add dummy styles
- Run through pipe

Commit:
Add responsive wrapper and styles to elements

- Aid testing responsive design of emojion bar

Commit:
Add basic wireframe styling

- Hover effect
- Implement flexbox

teesloane commented:
This is EXCELLENT! Looks great @ChristofLee . Let me try and answer your questions:

> How many icons will be displayed?

I think we're going to cap it at 4/5 for now.

> Is there a set number of icons displayed (e.g. 4) or will users be able to add more icons?

This is an interesting idea. I hadn't thought of it much. This could be a possible cool v2.0 idea to explore. For now, we'll cap it at 4/5



> if the above is true, will there be a button to add more icons?

n/a


> Am I permitted to edit the classes and markup here? e.g. add or tags, use BEM classes

Yep! Don't worry too much about allowed when it comes to small stuff — if it breaks something , our tests should tell, or i'll make a comment in PR. for bigger things, it's good to check in with other collaborators :)

> Is this intended to be styled easily when added into a site? or should our styles take priority? e.g. color, font

This is a good question. Again I didn't think of this. I think for now we will have a default theme, but OH—themes would be pretty cool actually. Something to think of it we have enough time for 2.0 -- for now I'll try and stop myself from doing all this feature creeping haha

Commit:
Snippet
- Restyle emojion bar based on comments in PR (limit of icons)
- Update classes (BEM) and remove whitespace between elements

Example
- Rewrite example to better display emojion bar with real-life example of a comments section
- BEM classes
- Test examples

Commit:
Cleanup styles and spacing

- Add responsive media query for comments

Commit:
Add borders to emojion bar to better fit wireframe

- Implement styles from old commit

Commit:
Add style-less test example

- Create comment with no classes

Commit:
Style container to better fit wireframe

- Responsive styles for container
- Add flexbox
- Make container full width like wireframe
- Add extra icon

ChristofLee commented:
Based on the comment above and my changes, I have updated my PR Summary:

## Summary
Basic styles for the Emojion bar including animated hover effects.

- Create responsive styles for the example.html doc
- Create responsive styles for the Emoji-bar
- Add style function to the pipe
- Load in styles for the Emoji-bar
- Create design examples
- BUG: Fix container being added to <style> tag

## Screenshots
Basic Emoji-bar:
<img width="900" alt="screen shot 2017-05-06 at 16 51 02" src="https://cloud.githubusercontent.com/assets/4659534/25773810/453226e8-327c-11e7-95ab-2194e04094c0.png">

## Checklist
- [x] I have updated the documentation to reflect my changes **N/A**
- [x] I have added tests to cover my changes (if relevant) **N/A**
- [x] All new and existing tests passed **PASSED**

teesloane commented:
Fantastic work @ChristofLee . Merging this now.
